Helper.py

```
@@ -0,0 +1,344 @@
+    center_offsets = (np.array(positions)+np.array([Lx,Ly,Lz])/2)
+    center_offsets[:,0] += (-tilt_factor*center_offsets[:,1])
+    # center_offsets = center_offsets.at[:,0].add(-tilt_factor*center_offsets.at[:,1].get())
+    center_offsets = center_offsets / np.array([Lx,Ly,Lz])  * np.array([Nx, Ny, Nz])
+    center_offsets -= np.array(center_offsets, dtype = int) #in grid units, not particle radius units
+    center_offsets = np.where(center_offsets>0.5, -(1-center_offsets), center_offsets)
+    for i in range(gauss_P):
+        for j in range(gauss_P):
+            for k in range(gauss_P):
+                grid[i, j, k, :] = gridh*gridh* ( (i - int(gauss_P/2) - center_offsets[:,0] + tilt_factor*(j - int(gauss_P/2) - center_offsets[:,1])   )**2 
+                                                 + (j - int(gauss_P/2) - center_offsets[:,1] )**2 
+                                                 + (k - int(gauss_P/2) - center_offsets[:,2] )**2) 
+    return grid
+
+
+
+
+def CreateRandomConfiguration(L, N, seed):
+    
+    def distance_periodic(p1, p2, L):
+        d = np.abs(p1 - p2)
+        d = np.where(d > L / 2, L - d, d)
+        return (np.sum(d*d))
+    
+    positions = np.zeros((N, 3), float)
+    max_attempts = 100000
+    attempts = 0
+    n = 0
+    random.seed(seed)
+
+    while n < N:
+        attempts += 1
+        x = random.uniform(-L / 2, L / 2)
+        y = random.uniform(-L / 2, L / 2)
+        z = random.uniform(-L / 2, L / 2)
+        overlap = False
+
+        for i in range(n):
+            d = distance_periodic(positions[i, :], np.array([x, y, z]), L)
+            if d < 4.5:  # The distance should be compared with the sum of the radii
+                overlap = True
+                break
+
+        if not overlap:
+            positions[n, :] = np.array([x, y, z])
+            n += 1
+            # print(n)
+
+        if attempts > max_attempts:
+            print('Computation too long, abort.')
+            break
```

This doesn't actually abort, right? Might this give a bad state?

---

Probably it's better to just throw an error.